notifier/notifier.go

```diff
@@ -610,12 +610,12 @@ func (n *Manager) sendAll(alerts ...*Alert) bool {
 			go func(ctx context.Context, client *http.Client, url string, payload []byte, count int) {
 				if err := n.sendOne(ctx, client, url, payload); err != nil {
 					level.Error(n.logger).Log("alertmanager", url, "count", count, "msg", "Error sending alert", "err", err)
-					n.metrics.errors.WithLabelValues(url).Inc()
+					n.metrics.errors.WithLabelValues(url).Add(float64(count))
 				} else {
 					numSuccess.Inc()
```

Comment:
I see the "equivalent" counter is still incremented once.
Also, for me the `metrics.errors` metric name and description do convey that the it tracks the failing sends (`sendOne`).
I think what should be adjusted is the alert, maybe a `rate() > 0` is sufficient to warn users are errors.

If we really care about the percentage, maybe we should expose another metric (or rename and change the purpose of  `metrics.errors` in 3.0)

Reply:
> I think what should be adjusted is the alert, maybe a `rate() > 0` is sufficient to warn users are errors.
> If we really care about the percentage


Yeah that's also an option.  The percentage gave some protection against a noisy alert, but maybe the `for: 15m` does the same job and we can stick to a `rate() > 0`. Are 3% of constant errors expected? Looks like a lot to me anyways 🤷 f04654b73be13381be5674f5b4559f8f6c711fad does the `rate() > 0`


> plus for real `ToAnyAlertmanager`, `dropped_total` is more appropriate

dropped_total isn't per-alertmanager and can also be incremented when alerts are dropped because the queue capacity is too small: [code](https://github.com/prometheus/prometheus/blob/b90dbdb08502dd157b077f14fcb4d553a28d0676/notifier/notifier.go#L420-L437). I think that's ok and we can use dropped_total instead of errors_total and change up the query a little bit to account for the missing `alertmanager` label. WDYT?

Reply:
I think it's fine to change from `Inc()` to `Add(float64(count))`. I don't see a reason why having more granularity would be bad here. I only see the net positive of being able to compute the ratio correctly. 

@beorn7, or some other maintainer seeing this, would this be considered a breaking change? I'm not exactly sure what the metrics guarantees are for this package.

Reply:
> dropped_total isn't per-alertmanager and can also be incremented when alerts are dropped because the queue capacity is too small: [code](https://github.com/prometheus/prometheus/blob/b90dbdb08502dd157b077f14fcb4d553a28d0676/notifier/notifier.go#L420-L437). I think that's ok and we can use dropped_total instead of errors_total and change up the query a little bit to account for the missing alertmanager label. WDYT?

After a closer look at `PrometheusErrorSendingAlertsToAnyAlertmanager` again I think `ToAny` is used to mean that it gives the error ratio across all altermanagers combined and not the ratio of sends that weren't received by ANY alertmanager. (I don't see why we don't have any alert with `dropped_total` though)

> I think it's fine to change from Inc() to Add(float64(count)). I don't see a reason why having more granularity would be bad here. I only see the net positive of being able to compute the ratio correctly.

We don't guarantee any stability over `/metrics` (plus 3.0 is around the corner). If the percentage is essential, let's adjust the metric description then and make sure it's in the changelog.

Reply:
Yeah, metrics are explicitly allowed to change without a major release (but hey, this _will_ go into a major release, if we hurry :).

(Just commenting on the formality of a breaking change. I haven't looked too deeply into the meaning of the metric, but you seem to have nailed that already.)